src/services/account.js

```diff
@@ -34,6 +34,28 @@ class AccountService {
             companyName: account.company.name,
         };
     }
+
+    async findAndDeleteByCompanyId(company) {
+        const account = await Account.findOne({
+            company
+        },
+        (err) => {
+            if (err) {
+                console.error(err);
+                throw err;
+            }
+        });
+        await Account.findByIdAndRemove(
+            account._id,
+            (err) => {
+                if (err) {
+                    console.error(err);
+                    throw err;
+                }
+            }
+        );
+        return account;
+    }
```

Comment:
I tried to simply call findByIdAndRemove but the function was returning null instead of the account, so I did it like this